workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/InterestQuery.cs

```diff
@@ -70,14 +78,47 @@ public InterestQuery FilterResults(uint resultComponentId, params uint[] resultC
             var resultIds = new List<uint>(resultComponentIds.Length + 1) { resultComponentId };
             resultIds.AddRange(resultComponentIds);
 
+            return FilterResults(resultIds);
+        }
+
+        /// <summary>
+        ///     Defines what components to return in the query results.
+        /// </summary>
+        /// <param name="resultComponentIds">
+        ///     Set of IDs of components to return from the query results.
+        /// </param>
+        /// <remarks>
+        ///     At least one component ID must be provided. Query results are not filtered
+        ///     if resultComponentIds is null or contains 0 elements.
+        /// </remarks>
+        /// <returns>
+        ///     An updated InterestQuery object.
+        /// </returns>
+        public InterestQuery FilterResults(IEnumerable<uint> resultComponentIds)
+        {
+            if (IsEnumerableNullOrEmpty(resultComponentIds))
+            {
+                Debug.LogWarning("At least one InterestQuery must be provided to filter a queries results.");
```

Comment:
Why do we throw exceptions in some places and only log warnings for others?

Reply:
We log warnings in `InterestTemplate` and `InterestQuery` because providing an IEnumerable with 0 elements can be safely continued with a no-op

However with `Constraint`, providing an IEnumerable with 0 elements into `Any` or `All` would mean that the constraint being set would be an empty list - which is equivalent to not having a constraint

The exception throwing is defensive in the case that an empty `QueryConstraint` is invalid to the runtime (chasing up with them to get clarity on this)

Reply:
@jamiebrynes7 got back from Runtime, if 0 Constraints are set then it is considered to be a user error